lib/active_merchant/billing/gateways/barclaycard_smartpay.rb

```diff
@@ -312,6 +315,14 @@ def store_request(options)
         hash[:shopperReference] = options[:customer] if options[:customer]
         hash.keep_if { |_, v| v }
       end
+
+      def add_threeds(post,options)
+        post[:additionalData] = {}
```

Comment:
This can be condensed to just

```
post[:additionalData] = {executeThreeD: 'true'}
post[:browserInfo] = {userAgent: options[:user_agent], acceptHeader: options[:accept_header]}
```